### PR TITLE
feat(payments-paypal): Create PayPalManager getCustomerPayPalSubscriptions

### PR DESCRIPTION
**Pull request by xlisachan**
## This pull request

- [x] creates PayPalManager getCustomerPayPalSubscriptions
  - [x] The method should take a stripe customer as a non-optional argument
  - [x] Return all customer subscriptions where collection method is send_invoice

## Issue that this pull request solves

Closes: FXA-8938

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.

### CODE REVIEW

**Review by StaberindeZA — APPROVED** [review]
r+. lgtm

File: libs/payments/paypal/src/lib/paypal.manager.ts [hunk 1]
```diff
@@ -35,6 +38,14 @@ export class PayPalManager {
     };
   }
 
+  async getCustomerPayPalSubscriptions(customer: Stripe.Customer) {
+    return customer.subscriptions?.data.filter(
```
Comment (StaberindeZA) [inline libs/payments/paypal/src/lib/paypal.manager.ts:42]: I left a similar comment on the other PR, but I just remembered that in the ADR 0040-stripe-typings, the team decided to never use expanded properties. I believe that applies to `customer.subscriptions` as well.

**Review by StaberindeZA — CHANGES_REQUESTED** [review]
After further investigation, looks like some changes might still be needed. Discussions on going.

File: libs/payments/stripe/src/lib/stripe.manager.ts [hunk 2]
```diff
@@ -19,12 +26,45 @@ export class StripeManager {
...
+   *
+   * @param customerId
+   */
+  async fetchActiveSubscriptions(subscriptions: Stripe.Subscription[]) {
+    return subscriptions.filter(
+      (sub) =>
+        ACTIVE_SUBSCRIPTION_STATUSES.includes(sub.status) &&
+        sub.collection_method === 'send_invoice'
+    );
+  }
```
Comment (julianpoy) [inline libs/payments/stripe/src/lib/stripe.manager.ts:49]: I think this belongs in the paypal manager - `send_invoice` is the collection method used for paypal subs.

File: libs/payments/stripe/src/lib/stripe.manager.ts [hunk 3]
```diff
@@ -19,12 +26,45 @@ export class StripeManager {
...
+
+  /**
+   * Retrieves customer subscriptions
+   *
+   * @param customerId
+   */
+  async fetchActiveCustomerSubscriptions(customerId: string) {
+    const customer = await this.fetchActiveCustomer(customerId, {
+      expand: ['subscriptions'],
```
Comment (julianpoy) [inline libs/payments/stripe/src/lib/stripe.manager.ts:58]: Expand is not allowed in SP3 (see ADR Reino linked)

File: libs/payments/stripe/src/lib/stripe.manager.ts [hunk 4]
```diff
@@ -19,12 +26,45 @@ export class StripeManager {
...
+    });
+    if (!customer.subscriptions?.data)
+      throw new CustomerNoSubscriptionsError(customerId);
```
Comment (julianpoy) [inline libs/payments/stripe/src/lib/stripe.manager.ts:61]: We can just return an empty array if the user has no subscriptions, no need to throw an error.

File: libs/payments/stripe/src/lib/stripe.manager.spec.ts [hunk 5]
```diff
@@ -26,6 +27,33 @@ describe('StripeManager', () => {
...
+
+  const mockPayPalSubscription3 = SubscriptionFactory({
+    collection_method: 'send_invoice',
+    status: 'trialing',
+  });
+
+  const mockStripeSubscription = SubscriptionFactory({
+    collection_method: 'charge_automatically',
+    status: 'active',
+  });
```
Comment (julianpoy) [inline libs/payments/stripe/src/lib/stripe.manager.spec.ts:48]: Factories should be used as close as possible to your tests, if possible. That way you're only creating the test data you absolutely need in a test, as well as guaranteeing no side-effects if one of your tests (or test subjects) happen to modify the data.

**Review by StaberindeZA — APPROVED** [review]
r+wc. Just 2 small comments, with nice to have changes, but non-blocking.

Thank you for the discussion on this ticket.

**Pull request merged**